system/include/emscripten/key_codes.h

```diff
@@ -0,0 +1,390 @@
+#define DOM_VK_F15                 0x7E
+#define DOM_VK_F16                 0x7F
+#define DOM_VK_F17                 0x80
+#define DOM_VK_F18                 0x81
+#define DOM_VK_F19                 0x82
+#define DOM_VK_F20                 0x83
+#define DOM_VK_F21                 0x84
+#define DOM_VK_F22                 0x85
+#define DOM_VK_F23                 0x86
+#define DOM_VK_F24                 0x87
+#define DOM_VK_NUM_LOCK            0x90
+#define DOM_VK_SCROLL_LOCK         0x91
+#define DOM_VK_WIN_OEM_FJ_JISHO    0x92
+#define DOM_VK_WIN_OEM_FJ_MASSHOU  0x93
+#define DOM_VK_WIN_OEM_FJ_TOUROKU  0x94
+#define DOM_VK_WIN_OEM_FJ_LOYA     0x95
+#define DOM_VK_WIN_OEM_FJ_ROYA     0x96
+#define DOM_VK_CIRCUMFLEX          0xA0
+#define DOM_VK_EXCLAMATION         0xA1
+#define DOM_VK_DOUBLE_QUOTE        0xA3
+#define DOM_VK_HASH                0xA3
+#define DOM_VK_DOLLAR              0xA4
+#define DOM_VK_PERCENT             0xA5
+#define DOM_VK_AMPERSAND           0xA6
+#define DOM_VK_UNDERSCORE          0xA7
+#define DOM_VK_OPEN_PAREN          0xA8
+#define DOM_VK_CLOSE_PAREN         0xA9
+#define DOM_VK_ASTERISK            0xAA
+#define DOM_VK_PLUS                0xAB
+#define DOM_VK_PIPE                0xAC
+#define DOM_VK_HYPHEN_MINUS        0xAD
+#define DOM_VK_OPEN_CURLY_BRACKET  0xAE
+#define DOM_VK_CLOSE_CURLY_BRACKET 0xAF
+#define DOM_VK_TILDE               0xB0
+#define DOM_VK_VOLUME_MUTE         0xB5
+#define DOM_VK_VOLUME_DOWN         0xB6
+#define DOM_VK_VOLUME_UP           0xB7
+#define DOM_VK_COMMA               0xBC
+#define DOM_VK_PERIOD              0xBE
+#define DOM_VK_SLASH               0xBF
+#define DOM_VK_BACK_QUOTE          0xC0
+#define DOM_VK_OPEN_BRACKET        0xDB
+#define DOM_VK_BACK_SLASH          0xDC
+#define DOM_VK_CLOSE_BRACKET       0xDD
+#define DOM_VK_QUOTE               0xDE
+#define DOM_VK_META                0xE0
+#define DOM_VK_ALTGR               0xE1
+#define DOM_VK_WIN_ICO_HELP        0xE3
+#define DOM_VK_WIN_ICO_00          0xE4
+#define DOM_VK_WIN_ICO_CLEAR       0xE6
+#define DOM_VK_WIN_OEM_RESET       0xE9
+#define DOM_VK_WIN_OEM_JUMP        0xEA
+#define DOM_VK_WIN_OEM_PA1         0xEB
+#define DOM_VK_WIN_OEM_PA2         0xEC
+#define DOM_VK_WIN_OEM_PA3         0xED
+#define DOM_VK_WIN_OEM_WSCTRL      0xEE
+#define DOM_VK_WIN_OEM_CUSEL       0xEF
+#define DOM_VK_WIN_OEM_ATTN        0xF0
+#define DOM_VK_WIN_OEM_FINISH      0xF1
+#define DOM_VK_WIN_OEM_COPY        0xF2
+#define DOM_VK_WIN_OEM_AUTO        0xF3
+#define DOM_VK_WIN_OEM_ENLW        0xF4
+#define DOM_VK_WIN_OEM_BACKTAB     0xF5
+#define DOM_VK_ATTN                0xF6
+#define DOM_VK_CRSEL               0xF7
+#define DOM_VK_EXSEL               0xF8
+#define DOM_VK_EREOF               0xF9
+#define DOM_VK_PLAY                0xFA
+#define DOM_VK_ZOOM                0xFB
+#define DOM_VK_PA1                 0xFD
+#define DOM_VK_WIN_OEM_CLEAR       0xFE
+
+static const char *emscripten_dom_vk_to_string(int dom_vk_code)
```

Comment:
We definitely cannot inline this, since the code size that this adds in the form of all the strings is huge. I'll look at turning this into a library instead.